scripts/monte_carlo.py

```diff
@@ -147,20 +192,145 @@ def monte_carlo_sampling_scipy(
     Documentation for Latin Hypercube: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.qmc.LatinHypercube.html#scipy.stats.qmc.LatinHypercube
     Orthogonal LHS is better than basic LHS: https://github.com/scipy/scipy/pull/14546/files, https://en.wikipedia.org/wiki/Latin_hypercube_sampling
     """
+
     sampler = qmc.LatinHypercube(
         d=N_FEATURES,
         centered=centered,
         strength=strength,
         optimization=optimization,
         seed=seed,
     )
+
     lh = sampler.random(n=SAMPLES)
+
+    lh = rescale_distribution(lh, DISTRIBUTION, DISTRIBUTION_PARAMS)
     discrepancy = qmc.discrepancy(lh)
-    logger.info(f"Hypercube discrepancy is: {discrepancy}")
+    logger.info(
+        "Discrepancy is:", discrepancy, " more details in function documentation."
+    )
 
     return lh
 
 
+def rescale_distribution(
+    latin_hypercube: np.array, distribution: str, distribution_params: list
```

Comment:
currently we don't have the hinting. It would be interesting to discuss on whether we should implement.
The discussion may extend to pypsa-eur as well

Reply:
oh...I can move this conversation to the discord channel

Reply:
I agree to initiate the discussion on this.

Reply:
That could be nice to discuss in the pypsa meets earth channel, ideally also during the pypsa-earth periodic meetings :)
That's supernice to hear the perspective of CS :)

Reply:
Awesome

Reply:
moved to the discussion :) resolving here